src/MyNUnit/MyNUnit/MyTestClass.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace MyNUnit;
+
+/// <summary>
+/// Processes custom tests from one testclass
+/// </summary>
+public class MyTestClass
+{
+    private readonly IEnumerable<MethodInfo> _testMethods;
+    private readonly IEnumerable<MethodInfo> _afterClassMethods;
+    private readonly IEnumerable<MethodInfo> _beforeClassMethods;
+    private readonly IEnumerable<MethodInfo> _afterMethods;
+    private readonly IEnumerable<MethodInfo> _beforeMethods;
+    private bool _needsToStop;
+    private readonly ConcurrentBag<TestState> _testStates = new();
+    private readonly Type _classType;
+
+    public MyTestClass(Type type)
+    {
+        _classType = type;
+        _testMethods = type.GetMethods()
+            .Where(m => m.GetCustomAttributes(typeof(Test), false).Length > 0 && IsMethodOk(m, false));
+        _afterMethods = type.GetMethods()
+            .Where(m => m.GetCustomAttributes(typeof(After), false).Length > 0 && IsMethodOk(m, false));
+        _beforeMethods = type.GetMethods()
+            .Where(m => m.GetCustomAttributes(typeof(Before), false).Length > 0 && IsMethodOk(m, false));
+        _afterClassMethods = type.GetMethods().Where(m =>
+            m.GetCustomAttributes(typeof(AfterClass), false).Length > 0 && IsMethodOk(m, true));
+        _beforeClassMethods = type.GetMethods().Where(m =>
+            m.GetCustomAttributes(typeof(BeforeClass), false).Length > 0 && IsMethodOk(m, true));
+    }
+
+    private static bool IsMethodOk(MethodInfo methodInfo, bool shouldBeStatic)
+    {
+        var isVoidAndParameterless = methodInfo.ReturnType == typeof(void) && methodInfo.GetParameters().Length == 0;
+        if (shouldBeStatic)
+        {
+            return isVoidAndParameterless && methodInfo.IsStatic;
+        }
+        return isVoidAndParameterless && !methodInfo.IsStatic;
```

Comment:
```suggestion
        return isVoidAndParameterless && (methodInfo.IsStatic == shouldBeStatic);
```
Обычно я против хитроумных логических формул, позволяющих сэкономить пару строк кода, но тут, кажется, это делает код actually более читаемым